src/otx/algorithms/anomaly/adapters/anomalib/strategies/xpu_single.py

```diff
@@ -0,0 +1,67 @@
+"""Lightning strategy for single XPU devic."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from typing import Optional
+
+import pytorch_lightning as pl
+import torch
+from lightning_fabric.plugins import CheckpointIO
+from lightning_fabric.utilities.types import _DEVICE
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies import StrategyRegistry
+from pytorch_lightning.strategies.single_device import SingleDeviceStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from otx.algorithms.common.utils.utils import is_xpu_available
+
+
+class SingleXPUStrategy(SingleDeviceStrategy):
+    """Strategy for training on single XPU device."""
+
+    strategy_name = "xpu_single"
+
+    def __init__(
+        self,
+        device: _DEVICE = "xpu:0",
+        accelerator: Optional["pl.accelerators.Accelerator"] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision_plugin: Optional[PrecisionPlugin] = None,
+    ):
+
+        if not is_xpu_available():
+            raise MisconfigurationException("`SingleXPUStrategy` requires XPU devices to run")
+
+        super().__init__(
+            accelerator=accelerator,
+            device=device,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+
+    @property
+    def is_distributed(self) -> bool:
+        """Returns true if the strategy supports distributed training."""
+        return False
+
+    def setup(self, trainer: "pl.Trainer") -> None:
+        """Sets up strategy."""
+        self.model_to_device()
+        super().setup(trainer)
+
+    def setup_optimizers(self, trainer: "pl.Trainer") -> None:
+        """Sets up optimizers."""
+        super().setup_optimizers(trainer)
+        model, optimizer = torch.xpu.optimize(trainer.model, optimizer=trainer.optimizers[0])
+        trainer.optimizers = [optimizer]
+        trainer.model = model
```

Comment:
I'm just wondering, is it okay if we don't use `self.optimizer`?
https://github.com/Lightning-AI/pytorch-lightning/blob/fc195b95405e9e2629466e5b28c6a9243209d596/src/pytorch_lightning/strategies/strategy.py#L142-L144

Reply:
`optimizers` setter of `trainer` sets optimizers to `strategy.optimizers`. https://github.com/Lightning-AI/pytorch-lightning/blob/fc195b95405e9e2629466e5b28c6a9243209d596/src/pytorch_lightning/trainer/trainer.py#L1742
so it works. But IMHO, using `self.optimizer` seems more clear. How do you think about, @sovrasov ?